### PR TITLE
Change existing DTO in Create Handlers to  Create DTO without id. Issue 1343

### PR DESCRIPTION
**Pull request by ValDekh**
## Summary of change

Change existing DTO in Create Handlers to  Create DTO without id

### CODE REVIEW

File: Streetcode/Streetcode.BLL/Mapping/Streetcode/TextContent/TermProfile.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/Mapping/Streetcode/TextContent/TermProfile.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.BLL/Mapping/Streetcode/TextContent/TextProfile.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/Mapping/Streetcode/TextContent/TextProfile.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Fact/Create/CreateFactCommand.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/Fact/Create/CreateFactCommand.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.BLL/MediatR/Sources/SourceLinkCategory/Update/UpdateCategoryCommand.cs [hunk 1]
```diff
@@ -5,5 +5,5 @@
 
 namespace Streetcode.BLL.MediatR.Sources.SourceLink.Update
 {
-    public record UpdateCategoryCommand(SourceLinkCategoryDTO Category) : IRequest<Result<Unit>>;
+    public record UpdateCategoryCommand(UpdateSourceLinkCategoryDTO Category) : IRequest<Result<UpdateSourceLinkCategoryDTO>>;
```
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/MediatR/Sources/SourceLinkCategory/Update/UpdateCategoryCommand.cs:8]: UpdateSourceLinkCategoryDTO or SourceLinkCategoryUpdateDTO? Same for every DTO that starts with Update/Create

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Media/Image/GetAllImageTests/GetAllImageTests.cs [hunk 2]
```diff
@@ -113,12 +113,12 @@ private List<ImageDTO> GetImagesDTOList()
...
-        private void MockRepositoryAndMapper(List<Image> ImageList, List<ImageDTO> ImageListDTO)
+        private void MockRepositoryAndMapper(List<DAL.Entities.Media.Images.Image> ImageList, List<ImageDTO> ImageListDTO)
         {
             _mockRepo.Setup(r => r.ImageRepository.GetAllAsync(
-            It.IsAny<Expression<Func<Image, bool>>>(),
-            It.IsAny<Func<IQueryable<Image>,
-            IIncludableQueryable<Image, object>>>()))
+            It.IsAny<Expression<Func<DAL.Entities.Media.Images.Image, bool>>>(),
+            It.IsAny<Func<IQueryable<DAL.Entities.Media.Images.Image>,
+            IIncludableQueryable<DAL.Entities.Media.Images.Image, object>>>()))
```
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Media/Image/GetAllImageTests/GetAllImageTests.cs:121]: We can create entity at the head of file like that: using ImageEntity = DAL.Entities.Media.Images.Image
And then just use ImageEntity

**Review by Lazy-Lenny — APPROVED** [review]

**Review by Spoilsport0 — APPROVED** [review]

**Review by vasylashka — COMMENTED** [review]
Change base branch from master to release/1.0.0

File: Streetcode/Streetcode.BLL/DTO/Sources/CreateSourceLinkCategoryDTO.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Streetcode.BLL.DTO.Sources
+{
+    public class CreateSourceLinkCategoryDTO
+    {
+        public int Id { get; set; }
```
Comment (yaakymiv) [inline Streetcode/Streetcode.BLL/DTO/Sources/CreateSourceLinkCategoryDTO.cs:5]: Having the same question as Vlad : why do we have sort of duplicate DTOs? And one more thing unclear: why some create DTOs have Id while some others do not?

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Fact/Create/CreateFactCommand.cs
Comment (ValDekh) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/Fact/Create/CreateFactCommand.cs:1]: Done

File: Streetcode/Streetcode.BLL/MediatR/Sources/SourceLinkCategory/Update/UpdateCategoryCommand.cs (hunk 1 above)
Comment (ValDekh) [reply Streetcode/Streetcode.BLL/MediatR/Sources/SourceLinkCategory/Update/UpdateCategoryCommand.cs]: SourceLinkCategoryDTO - it's an old variant of respond/respond DTO and it had been before I merged my branch with last changes from develop. Now current variant is with UpdateDTO and all last unit tests are set and passed with UpdateDTO. In my opinion, there should be another one ticket for check all UpdateHadlers for MediatR. So in some cases there is Update DTO but at the same time there different Hadlers which utilize initial DTO.  So the matter in another ticket.

File: Streetcode/Streetcode.BLL/DTO/Sources/CreateSourceLinkCategoryDTO.cs (hunk 3 above)
Comment (ValDekh) [reply Streetcode/Streetcode.BLL/DTO/Sources/CreateSourceLinkCategoryDTO.cs]: I deleted all unnecessary DTO

**Pull request merged**